src/NLog/Targets/FileTarget.cs

```diff
@@ -70,6 +70,13 @@ public class FileTarget : TargetWithLayoutHeaderAndFooter, ICreateFileParameters
         private Timer autoClosingTimer;
         private int initializedFilesCounter;
 
+        /// <summary>
+        /// Provides the shape of an archive file changed event. Event name will describe the action.
+        /// </summary>
+        /// <param name="archiveFilename">The name of the file in question.</param>
+        public delegate void ArchiveFileAlteredHandler(string archiveFilename);
+        
```

Comment:
With archiving, we like to know to things: the original path and the new archive File Path. This will opt for two separate delegates.